File: src/routes/domains.js
/**
 * @swagger
 * tags:
 *   name: Domains
 *   description: FusionPBX domains (tenants)
 */

const express = require('express');
const { authenticate } = require('../middleware/auth');
const dbService = require('../services/dbService');

const router = express.Router();
router.use(authenticate);

/**
 * @swagger
 * /api/domains:
 *   get:
 *     summary: List all FusionPBX domains
 *     tags: [Domains]
 *     security:
 *       - ApiKeyAuth: []
 *       - BearerAuth: []
 *     responses:
 *       200:
 *         description: Domain list
 */
router.get('/', async (req, res, next) => {
  try {
    const domains = await dbService.getDomains();
    res.json({ domains, count: domains.length });
  } catch (err) {
    next(err);
  }
});

module.exports = router;

File: python/routers/domains.py
from fastapi import APIRouter, Depends
from deps.auth import require_auth
from services.db_service import db_service

router = APIRouter(prefix='/api/domains', tags=['Domains'])


@router.get('/')
async def get_domains(_user=Depends(require_auth)):
    # User-key auth can only see their own domain
    if _user.get('source') == 'user':
        user_domain = _user.get('domain')
        if user_domain:
            return {'domains': [{'domain_name': user_domain}], 'count': 1}
    domains = await db_service.get_domains()
    return {'domains': domains, 'count': len(domains)}
